zerver/middleware.py

```diff
@@ -420,7 +420,11 @@ def process_response(
         if requestor_for_logs is None:
             # Note that request.user is a Union[RemoteZulipServer, UserProfile, AnonymousUser],
             # if it is present.
-            if hasattr(request.user, "format_requestor_for_logs"):
+            if remote_server := request_notes.remote_server:
+                requestor_for_logs = remote_server.format_requestor_for_logs()
+            # "user" might not be set when the request did not make it to Django's
+            # AuthenticationMiddleware due to any error being raised in the middlewares chain.
```

Comment:
Is this still a thing that can happen, given that this middleware runs after `AuthenticationMiddleware`?

Reply:
It does not. I removed this comment and the one above it to reflect that.